src/libasr/codegen/asr_to_c_cpp.h

```diff
@@ -691,11 +695,15 @@ R"(#include <stdio.h>
         } else {
             src += "(" + left + ")";
         }
-        src += ASRUtils::cmpop_to_str(x.m_op);
-        if (right_precedence <= last_expr_precedence) {
-            src += right;
+        if( T::class_type == ASR::exprType::StringCompare ) {
+            src = "strcmp(" + left + ", " + right + ") == 0";
```

Comment:
This change makes it works for C, but breaks it for C++, which does allow "==" for `std::string`.